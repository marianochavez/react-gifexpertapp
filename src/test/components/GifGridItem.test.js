import React from 'react';
import {shallow} from "enzyme";
import {GifGridItem} from "../../components/GifGridItem";


describe("Test GifGridItem", () => {

    const title = "test title";
    const url = "https://asd.com";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    
    test('should show GifGridItem correctly', () => {

        expect(wrapper).toMatchSnapshot();
    })
    
    test('should have a paragraph with title', () => {
        
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    })
    
    test('should have a img with url', () => {

        const img = wrapper.find("img");
        // console.log(img.html());
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    })

    test('should have animate__fadeIn', () => {
        
        const div = wrapper.find("div");
        expect(div.hasClass("animate__fadeIn")).toBe(true);
    })
    


});