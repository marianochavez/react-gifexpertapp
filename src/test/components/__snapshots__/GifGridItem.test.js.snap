// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGridItem should show GifGridItem correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="test title"
    src="https://asd.com"
  />
  <p>
    test title
  </p>
</div>
`;
